Template.php

```diff
@@ -738,8 +738,10 @@ protected function get_identifiers_from_url($url_sent = NULL) {
     if ($doi = extract_doi($url)[1]) {
       if (is_null($url_sent)) {
         if (doi_active($doi)) {
-          report_forget("Recognized DOI in URL; dropping URL");
-          $this->forget('url');
+          if ((mb_strpos(strtolower($url), ".pdf") === FALSE) || (str_ireplace($url, '', PUBLISHER_WEBSITES) != $url)) {
```

Comment:
I can't follow the syntax of str_ireplace here.  Can we include a testcase, so that it's clear that this works?

Reply:
I will add a test case.   The logic is very odd, but it works.   The logic is simple but obscure.   You look for the publisher websites and replace them.   Then you look and see if the url changed, you know you found one.

Reply:
Should the code read `str_ireplace(PUBLISHER_WEBSITES, '', $url)`?

Reply:
Ooops.  All fixed now.  Test added too.  Closing this conversation.  Not saying it is fixed, but it should now be obvious if test fails.